Frameworks/Foundation/NSOperationQueue.mm

```diff
@@ -220,65 +369,19 @@ - (NSOperationQueue*)_doMainWork {
 /**
  @Status Interoperable
 */
-- (BOOL)hasMoreWork {
-    if (priv->curOperation != nil) {
-        return TRUE;
-    }
-
-    for (int i = 0; i < NSOperationQueuePriority_Count; i++) {
-        if (priv->queues[i] != NULL) {
-            return TRUE;
-        }
-    }
-
-    return FALSE;
-}
-
-/**
- @Status Interoperable
-*/
 - (id)init {
     if (self = [super init]) {
-        priv = new NSOperationQueuePriv();
-
-        priv->_maxConcurrentOperationCount = 1;
-        priv->workAvailable = [[NSCondition alloc] init];
-        priv->suspendedCondition = [[NSCondition alloc] init];
-        priv->allWorkDone = [[NSCondition alloc] init];
-        priv->isSuspended = 0;
-
         _completionQueue = dispatch_queue_create("NSOperation finish queue", nullptr);
+        _priv = std::make_unique<NSOperationQueuePriv>(self);
```

Comment:
If the no-argument constructor on NSOperationQueuePriv does what we need, we don't need to indirect through a  unique_ptr and can embed it directly.